example/api.rsh

```diff
@@ -1,6 +1,8 @@
-/// Prints a friendly greeting to stdout.
-fn hello_world();
 
-/// Returns a future that prints a friendly
-/// greeting to stdout.
-fn async_hello_world() -> Future<Result<u8>>;
+fn get_image() -> buffer;
+
+fn create(n: usize) -> DataTest;
+object DataTest {
+    fn get_copy() -> Vec<u8>;
+    fn get_shmem() -> buffer;
```

Comment:
let's make `buffer` typed, too, please `buffer<u8>` and maybe we could figure out a buffer for a custom type as well? `buffer<DataTest>`?

Reply:
This buffer type has nothing to do with the discussion we had about lists. It's just a blob of data (zerocopybuffer). I'm afraid it is impossible to make this type generic since dart doesn't allow programmers to control the alignment of structs and lists. It should be possible to parse the struct definitions and do translations manually but that's not how ffi-gen works.